recipes/elfio/all/test_package/example.cpp

```diff
@@ -0,0 +1,14 @@
+
+#include <elfio/elfio.hpp>
+#include <cstdlib>
+
+using namespace ELFIO;
+
+int main() {
+    // just check we can create an reader, that means the recipe works
+    elfio reader;
+    if ( !reader.load( "/does/not/exist" ) ) {
+        return EXIT_SUCCESS;
+    }
```

Comment:
Maybe if `__linux__` it could read itself `argv[0]` and print all contents using `elfio_dump.hpp`

```suggestion
    }
#ifdef __linux__
    std::string target = argv[0];

    if (!reader.load(target)) {
        std::cerr << "error: not an ELF file\n";
        return 1;
    }

    dump::header         ( std::cout, reader );
    dump::section_headers( std::cout, reader );
    dump::segment_headers( std::cout, reader );
    dump::symbol_tables  ( std::cout, reader );
    dump::notes          ( std::cout, reader );
    dump::dynamic_tags   ( std::cout, reader );
    dump::section_datas  ( std::cout, reader );
    dump::segment_datas  ( std::cout, reader );
#endif
```

Reply:
I don not see a very huge value in adding that complexity.
All that needs to be tested is that the header is found, and that is done. So test of the package works fine, we are not going to test elfio itself, this is done by elfio